core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -132,7 +132,7 @@ trait Endpoint[A] { self =>
     new Endpoint[B] {
       final def apply(input: Input): Endpoint.Result[B] = self(input) match {
         case EndpointResult.Matched(rem, trc, out) =>
-          EndpointResult.Matched(rem, trc, Rerunnable.fromFuture(fn(out.run)))
+          EndpointResult.Matched(rem, trc, out.flatMap(a => Task.async(fn(Future.value(a)))))
```

Comment:
this seems to be the cause

Reply:
Oh cool. Yeah, maybe it's fine to bounce thorough Future/Scheduler here as it's pretty close to what we're doing right now.